### DISCUSSION TITLE
TensorBoardLogger() gives "AttributeError: can't set attribute" error

### DISCUSSION THREAD

**Discussion by nkay28** [Lightning-AI/pytorch-lightning#16322] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, 
I'm trying to build a class with pl.LightningModule as super for training storm images using resnet (following an example on GitHub). 
I keep getting this 'AttributeError: can't set attribute' in the TensorBoard logs line. The next lines are for datapull from another class and I don't suspect the error to be there, but not sure at this point. 

` ```
class PretrainedWindModel(pl.LightningModule):
    def __init__(self, hparams):
        super(PretrainedWindModel, self).__init__()

        # self.hparams = hparams
        self.hparams.update(dict(hparams))
        self.learning_rate = self.hparams.get("lr", 2e-4)
        self.hidden_size = self.hparams.get("embedding_dim", 50)
        self.dropout = self.hparams.get("dropout", 0.1)
        self.max_epochs = self.hparams.get("max_epochs", 1)
        self.num_workers = self.hparams.get("num_workers", 0)
        self.batch_size = self.hparams.get("batch_size", 10)
        self.x_train = self.hparams.get("x_train")
        self.y_train = self.hparams.get("y_train")
        self.x_val = self.hparams.get("x_val")
        self.y_val = self.hparams.get("y_val")
        self.num_outputs = 1  # One prediction for regression

        # Where final model will be saved
        self.output_path = Path.cwd() / self.hparams.get("output_path", "model-outputs")
        self.output_path.mkdir(exist_ok=True)

        # Where TensorBoard logs will be saved
        self.log_path = Path.cwd() / self.hparams.get("log_path", "logs")
        self.log_path.mkdir(exist_ok=True)
        # self.logger = pl.loggers.TensorBoardLogger(self.log_path, name="benchmark_model")
        self.logger = TensorBoardLogger(self.log_path, name="benchmark_model")

        # Instantiate training and validation datasets
        self.train_dataset = DatasetWIND(self.x_train, self.y_train)
        self.val_dataset = DatasetWIND(self.x_val, self.y_val)
        self.model = self.prepare_model()
``` `

Error:

````
 AttributeError                            Traceback (most recent call last)
Input In [72], in <cell line: 1>()
----> 1 storm_model = PretrainedWindModel(hparams=hparams)
      2 storm_model.fit()

Input In [69], in PretrainedWindModel.__init__(self, hparams)
     25 self.log_path.mkdir(exist_ok=True)
     26 #self.logger = pl.loggers.TensorBoardLogger(self.log_path, name="benchmark_model")
---> 27 self.logger = TensorBoardLogger(self.log_path, name="benchmark_model")
     30 # Instantiate training and validation datasets
     31 self.train_dataset = DatasetWIND(self.x_train, self.y_train)

File C:\Users\Nitin_Kumar\anaconda3\lib\site-packages\torch\nn\modules\module.py:1255, in Module.__setattr__(self, name, value)
   1253     buffers[name] = value
   1254 else:
-> 1255     object.__setattr__(self, name, value)

AttributeError: can't set attribute 
````

**Comment by Giuseppe-1993** [comment]
Hi @nkay28,

I am experiencing the exact same problem when attempting to set the TensorBoardLogger as a class attribute. Did you find a solution?

**Comment by Xnhyacinth** [comment]
the same error
